linkerd/app/outbound/src/http/logical.rs

```diff
@@ -47,55 +47,60 @@ where
     let watchdog = cache_max_idle_age * 2;
 
     svc::stack(endpoint.clone())
-        .check_new_service::<Endpoint, http::Request<http::BoxBody>>()
         .push_on_response(
             svc::layers()
-                .push(svc::layer::mk(svc::SpawnReady::new))
+                .push(http::BoxRequest::layer())
                 .push(
                     metrics
                         .stack
                         .layer(stack_labels("http", "balance.endpoint")),
                 )
-                .push(http::BoxRequest::layer()),
+                // Ensure individual endpoints are driven to readiness so that
+                // the balancer need not drive them all directly.
+                .push(svc::layer::mk(svc::SpawnReady::new)),
```

Comment:
TIOLI: could be nice to add a `SpawnReady::layer`?

Reply:
https://github.com/tower-rs/tower/pull/536